subctl/issue-259-connect.md

```diff
@@ -0,0 +1,130 @@
+including component selection, Globalnet configuration etc., apart
+from those related to `broker-info.subm`. Since a single invocation of
+`connect` has knowledge of more than one cluster, some additional
+checks can be performed; in particular, the cluster CIDRs can be
+checked for overlaps, with the following behaviour:
+
+* in the default setup (no Globalnet configuration), Globalnet is
+  automatically enabled if the CIDRs overlap;
+* if Globalnet is explicitly disabled, and the CIDRs overlap,
+  `connect` fails with an appropriate error.
+
+Whether the broker is chosen automatically or specified using
+`--broker-context`, if it is inaccessible from any of the clusters,
+`connect` fails with an appropriate error, without changing any of the
+configuration on any of the clusters.
+
+### Connecting to an existing Submariner deployment
+
+If any of the provided contexts point to a cluster with an existing
+Submariner deployment, all such contexts are used to find the
+corresponding brokers. If more than one broker is found, `connect`
+fails with an appropriate error.
+
+If only one broker is found, connection continues using that broker.
+If the resulting setup is determined to be broken (Globalnet disabled
+with a CIDR conflict involving one or more of the additional
+clusters), `connect` fails with an appropriate error.
+
+If the broker is inaccessible from any of the additional clusters,
+`connect` fails with an appropriate error, without changing any of the
+configuration on any of the clusters.
+
+If all the provided contexts are already connected, `connect` doesn’t
+do anything and exits successfully.
+
+If a broker is found indirectly (that is to say, it isn’t listed in
+the contexts to be connected), and the cluster hosting it isn’t
+connected, that cluster’s configuration isn’t changed. This allows
+hub-spoke style setups, with a broker that isn’t part of the connected
+clusters, to be extended using `connect`. If the broker’s context is
+listed in the contexts to be connected, the cluster is connected if
+necessary.
+
+### Detailed operations
+
+The above considerations result in this sequence of steps:
+
+1. Connect to all the provided contexts, looking for Submariner
+   deployments. If a Submariner broker is found, remember it; if a
+   Submariner deployment is found, determine the corresponding broker
+   and remember it.
+
+2. If more than one broker is found in the above step, fail.
+
+3. If no broker is found, run connectivity tests from each cluster to
+   all the other clusters’ API server. If any cluster is not
+   accessible from another cluster, remove it from the set of broker
+   candidates. If the set of broker candidates ends up empty, fail.
+
+4. If a broker is found, run connectivity tests from each additional
+   cluster to its API server. If any of the tests fail, fail.
+
+5. Check for overlapping CIDRs among the clusters. If any overlap:
+
+   1. If an existing broker was found, and Globalnet isn’t enabled,
```

Comment:
In this case we should also ensure the new clusters don't overlap with any existing clusters already connected via the broker.